src/profile/hooks/useProfileDataEditor.ts

```diff
@@ -96,14 +102,31 @@ export function useProfileDataEditor({
     triggerUpdate();
     return executeMutationUpdateAndHandleResult(formValues, item.id);
   };
+
   const reset: ProfileDataEditorReturnType['reset'] = item => {
     const success = resetItem(item);
     success && triggerUpdate();
   };
 
+  const add = () => {
+    addItem();
+    triggerUpdate();
+  };
+
+  const hasNew = () => hasNewItem();
+
+  const remove = async (item: EditData) => {
+    removeItem(item);
+    triggerUpdate();
+    return Promise.resolve();
+  };
+
   return {
     editDataList: getEditData(),
     save,
     reset,
+    add,
+    hasNew,
```

Comment:
Could we inline the hasNewItem here?

Reply:
Do you mean adding 

`hasNew: ()=>hasNewItem?`
 
It is not allowed by sonarCube, I think. It wont allow mixing shorthands with other code in "return {...}" blocks.

But I changed the code where hasNewItem is created to 

`hasNewItem: hasNew,`

so "const hasNew = () => hasNewItem();"  is removed.

Reply:
Done.